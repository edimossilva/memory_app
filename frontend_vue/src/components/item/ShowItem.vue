<template>
  <div>
    <div class="card" :data-cy="`show_item__card_${item.key}`">
      <div class="card-content">
        <div class="content">{{ item.key }}: {{ item.value }}</div>
      </div>
      <footer class="card-footer">
        <a
          class="card-footer-item"
          :dataId="item.key"
          @click.prevent="onCopyButtonClick"
        >
          Copy
        </a>
        <a
          class="card-footer-item has-text-danger"
          :data-cy="`show_item__delete_button_${item.key}`"
          @click.prevent="onRemoveButtonClick"
        >
          Remove
        </a>
      </footer>
    </div>
  </div>
</template>

<script>
import { mapActions } from "vuex";
import { deleteItemApi } from "../../services/itemsApi";

export default {
  name: "ShowItem",
  props: {
    item: { type: Object, required: true },
  },
  components: {},
  methods: {
    ...mapActions(["removeItem"]),
    onRemoveButtonClick() {
      deleteItemApi(this.item.id).then(
        () => {
          this.removeItem(this.item);
        },
        (error) => {
          // this.showError();
          console.log(error.response.data.error_message);
        }
      );
    },
    onCopyButtonClick() {
      const { item, $copyText, $buefy } = this;
      $copyText(item.value).then(() => {
        $buefy.dialog.alert({
          message: `"${item.value}" copied to clipboard :)`,
          type: "is-primary",
          ariaRole: "alertdialog",
          ariaModal: true,
        });
      });
    },
  },
};
</script>
