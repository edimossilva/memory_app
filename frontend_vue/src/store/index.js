import Vue from "vue";
import Vuex from "vuex";

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    items: [],
  },
  mutations: {
    addItem(state, item) {
      state.items.push(item);
    },
    removeItem(state, item) {
      const index = state.items.findIndex((i) => i.key === item.key);
      state.items.splice(index, 1);
    },
    setItems(state, items) {
      state.items = items;
    },
    // mutate(state, payload) {
    //   state[payload.property] = payload.value;
    // },
  },
  actions: {
    addItem({ commit }, item) {
      commit("addItem", { ...item });
    },
    removeItem({ commit }, item) {
      commit("removeItem", item);
    },
    setItems({ commit }, items) {
      commit("setItems", items);
    },
  },
  modules: {},
});
