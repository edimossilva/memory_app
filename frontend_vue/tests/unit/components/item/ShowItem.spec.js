import { shallowMount } from "@vue/test-utils";
import ShowItem from "@/components/item/ShowItem.vue";
import Vue from "vue";
import VueRouter from "vue-router";
import store from "@/store/index";

Vue.use(VueRouter);

const item = { key: "mockKey1", value: "mockValue1" };

const spyRemoveItem = jest.spyOn(ShowItem.methods, "removeItem");

describe("ListItems.vue", () => {
  it("Matchs Snapshot", () => {
    const wrapper = shallowMount(ShowItem, {
      store,
      propsData: { item },
    });
    expect(wrapper).toMatchSnapshot();
  });

  describe("methods", () => {
    describe("onRemoveButtonClick()", () => {
      const wrapper = shallowMount(ShowItem, {
        store,
        propsData: { item },
      });

      beforeEach(() => {
        wrapper.vm.onRemoveButtonClick();
      });

      it("should call push to dashboard", () => {
        expect(spyRemoveItem).toHaveBeenCalledWith(item);
      });
    });

    // const wrapper = shallowMount(CreateItemModal, {
    //   store,
    //   stubs: {
    //     modal: true,
    //   },
    //   mocks: {
    //     $modal,
    //   },
    // });

    const $copyText = () => {};
    describe("onCopyButtonClick()", () => {
      const wrapper = shallowMount(ShowItem, {
        store,
        propsData: { item },
        mocks: { $copyText },
      });

      beforeEach(() => {
        wrapper.vm.onCopyButtonClick();
      });
    });
  });
});
