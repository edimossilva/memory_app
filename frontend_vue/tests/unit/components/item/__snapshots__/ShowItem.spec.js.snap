// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItems.vue Matchs Snapshot 1`] = `
<div>
  <div data-cy="show_item__card_mockKey1" class="card">
    <div class="card-content">
      <div class="content">mockKey1: mockValue1</div>
    </div>
    <footer class="card-footer"><a dataid="mockKey1" class="card-footer-item">
        Copy
      </a> <a data-cy="show_item__delete_button_mockKey1" class="card-footer-item has-text-danger">
        Remove
      </a></footer>
  </div>
</div>
`;
